From 9fda11ad1f7630e0c64cfe3a2203ca1f24525512 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marcus=20M=C3=BCller?= <marcus.mueller@ettus.com>
Date: Sun, 25 Oct 2015 11:25:33 +0100
Subject: [PATCH 13/14] docs: Install info updates

- fixed win boost bin link
- formatting
-  added detailed Win Build instructions
---
 host/docs/build.dox | 23 +++++++++++++----------
 1 file changed, 13 insertions(+), 10 deletions(-)

diff --git a/host/docs/build.dox b/host/docs/build.dox
index 0e722e3..05e4ff7 100644
--- a/host/docs/build.dox
+++ b/host/docs/build.dox
@@ -28,7 +28,7 @@ The following compilers are known to work and officially supported:
 
 - GCC >= 4.4
 - Clang >= 3.1
-- MSVC >= 2012
+- MSVC >= 2012; the free <a href="https://www.visualstudio.com/en-us/products/visual-studio-express-vs.aspx">Visual Studio Express Edition for Desktop</a> works.
 
 Other compilers (or lower versions) may work, but are unsupported.
 
@@ -45,7 +45,7 @@ Other compilers (or lower versions) may work, but are unsupported.
 - **Minimum Version:** 1.46
 - **Usage:** build time + runtime (required)
 - **Download URL:** http://www.boost.org/users/download/
-- **Download URL (Windows installer):** http://www.boostpro.com/download
+- **Download URL (Windows installer):** http://sourceforge.net/projects/boost/files/boost-binaries/
 
 ### LibUSB
 
@@ -68,9 +68,7 @@ Other compilers (or lower versions) may work, but are unsupported.
 - **Minimum Version:** 0.5.0
 - **Usage:** build time (required)
 - **Download URL:** http://www.makotemplates.org/download.html
-
-**Alternative method:**
-You can use `pip` or `easy_install` to install Mako from PyPi.
+- **Alternative method:** You can use `pip` or `easy_install` to install Mako from PyPi.
 
 ### Doxygen
 
@@ -78,10 +76,7 @@ You can use `pip` or `easy_install` to install Mako from PyPi.
 - **Minimum Recommended Version:** 1.8
 - **Usage:** build time (optional)
 - **Download URL:** http://www.stack.nl/~dimitri/doxygen/download.html#latestsrc
-
-**Alternate method:**
-Install **setuptools**, and use the **easy_install** command to install Docutils.
-http://pypi.python.org/pypi/setuptools
+- **Alternate method:** Install **setuptools**, and use the **easy_install** command to install Docutils. http://pypi.python.org/pypi/setuptools
 
 ### Git
 
@@ -180,9 +175,17 @@ or add it to `/etc/ld.so.conf` and make sure to run:
 - Set the path to the build directory: `<uhd-repo-path>/host/build`.
 - Make sure that the paths do not contain spaces.
 - Click "Configure" and select "Microsoft Visual Studio 10".
-- Set the build variables and click "Configure" again.
+- Set the build variables
+  - `Boost_INCLUDE_DIR` should point to the `PATH` where the Boost .hpp files are, e.g. `C:\local\boost_version\boost`
+  - `Boost_LIBRARY_DIR` should point to the pre-built libraries, e.g. `C:\local\boost_version\lib64-msvc-14.0`)
+  .
 - Click "Generate", and a project file will be created in the build directory.
 
+\subsection msvc_generate Compile and Install with Visual Studio
+- Open the `<uhd-repo-path>\\host\\build\\ALL_BUILD.vcxproj` in visual studio, generate the project. Watch the output console for errors.
+- If you did not specify a `CMAKE_INSTALL_PREFIX` where your user has write privileges, you must close Visual Studio, run it again with Administrator Privileges, and open `ALL_BUILD.vcxproj`.
+- Right click on the **INSTALL** project in the project explorer, generate it. Watch the output console for errors.
+
 \subsection libusb_notes LibUSB notes
 
 On Windows, CMake does not have the advantage of `pkg-config`,
-- 
2.1.4

